src/utils.rs

```diff
@@ -0,0 +1,39 @@
+//! Some useful utilities for the library
+
+///
+/// converts oid-string like "1.3.6" into vector Vec<u32>[1, 3, 6, 0]
+///
+/// # Examples
+/// ```
+/// use snmp::utils::oid_str_to_vec32;
+///
+/// assert_eq!(oid_str_to_vec32("1.3.6"), Ok(vec![1, 3, 6, 0]));
+///
+/// // skip a leading dot
+/// assert_eq!(oid_str_to_vec32(".1.3.6"), Ok(vec![1, 3, 6, 0]));
+///
+/// // do not add extra zero at end
+/// assert_eq!(oid_str_to_vec32("1.3.6.0"), Ok(vec![1, 3, 6, 0]));
+///
+/// // wrong oid
+/// assert!(oid_str_to_vec32("1.3.6.").is_err());
+/// ```
+pub fn oid_str_to_vec32(oid: &str) -> Result<Vec<u32>, String> {
```

Comment:
I'd call it `parse_oid`. And the error must be some error type, not a string. Like SnmpError or some other existing or new type.

Reply:
Do you want me to introduce a standard type Oid in the main crate file 'lib.rs' to make it public to everyone?

Reply:
Yes, make a struct `Oid`.
Implement `FromStr` trait from it. Once implemented you can write something like `let oid = "1.2.3".parse::<Oid>()`.
Implement `AsRef` trait for it.
Make it derived from debug and clone traits.